src/app/features/courses/components/courses-list/components/courses-search/courses-search.component.ts

```diff
@@ -1,21 +1,51 @@
-import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+import {
+  AfterViewInit,
+  Component,
+  ElementRef,
+  EventEmitter,
+  OnDestroy,
+  Output,
+  ViewChild,
+} from '@angular/core';
+import { fromEvent, map, filter, debounceTime, Subject, takeUntil } from 'rxjs';
 
 @Component({
   selector: 'app-courses-search',
   templateUrl: './courses-search.component.html',
   styleUrls: ['./courses-search.component.scss'],
 })
-export class CoursesSearchComponent implements OnInit {
-  searchValue: string = '';
+export class CoursesSearchComponent implements AfterViewInit, OnDestroy {
+  private readonly destroy$ = new Subject<boolean>();
 
   @Output()
   searchValueChange = new EventEmitter<string>();
 
+  @ViewChild('searchInput')
+  searchInput!: ElementRef;
+
   constructor() {}
 
-  ngOnInit(): void {}
+  ngAfterViewInit(): void {
+    if (this.searchInput.nativeElement) {
+      fromEvent(this.searchInput.nativeElement, 'keyup')
+        .pipe(
+          map((e: any) => {
```

Comment:
Try to write code as one-liner, if you don't need to handle several actions in your operators:

`map((e: any ) => (e.target as HTMLInputElement).value)`